### PR TITLE
feat: payment intent full flow

### PR DESCRIPTION
**Pull request by Andrey-Lavnik**

### CODE REVIEW

File: microservices/payment/src/entities/connect-account.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Allow, Length } from 'class-validator';
+import { Entity, Index, PrimaryColumn } from 'typeorm';
+
+@Entity()
+class ConnectAccount {
```
Comment (MatthewPattell) [inline microservices/payment/src/entities/connect-account.ts]: remove

File: microservices/payment/src/methods/index.ts [hunk 2]
```diff
@@ -28,6 +30,10 @@ export default (ms: Microservice): void => {
       ...CrudCustomer,
       create: CustomerCreate,
     },
+    connectAccount: {
```
Comment (MatthewPattell) [inline microservices/payment/src/methods/index.ts]: remove

File: microservices/payment/src/methods/stripe/connect-account/create.ts
Comment (MatthewPattell) [inline microservices/payment/src/methods/stripe/connect-account/create.ts:1]: microservices/payment/src/methods/stripe/connect-account.ts

File: microservices/payment/src/methods/stripe/connect-account/create.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import { Endpoint } from '@lomray/microservice-helpers';
+import { Type } from 'class-transformer';
+import { IsObject, IsString } from 'class-validator';
+import { getManager } from 'typeorm';
+import ConnectAccount from '@entities/connect-account';
+import Factory from '@services/payment-gateway/factory';
+
+class ConnectAccountCreateInput {
```
Comment (MatthewPattell) [inline microservices/payment/src/methods/stripe/connect-account/create.ts]:
```suggestion
class ConnectAccountInput {
```

File: microservices/payment/src/methods/stripe/connect-account/create.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  @IsString()
+  userId: string;
+}
+
+class ConnectAccountCreateOutput {
```
Comment (MatthewPattell) [inline microservices/payment/src/methods/stripe/connect-account/create.ts]:
```suggestion
class ConnectAccountOutput {
```

File: microservices/payment/src/methods/stripe/connect-account/crud.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { Endpoint } from '@lomray/microservice-helpers';
+import { getRepository } from 'typeorm';
+import ConnectAccount from '@entities/connect-account';
+
+/**
+ * CRUD controller for ConnectAccount entity
+ */
+const crud = Endpoint.controller(() => getRepository(ConnectAccount), {
```
Comment (MatthewPattell) [inline microservices/payment/src/methods/stripe/connect-account/crud.ts]: remove

File: microservices/payment/src/services/payment-gateway/abstract.ts [hunk 6]
```diff
@@ -82,6 +88,33 @@ abstract class Abstract {
...
+  public async createConnectAccount(userId: string, connectAccountId: string = uuid()) {
+    const connectAccount = this.connectAccountRepository.create({
+      connectAccountId,
+      userId,
+    });
+
+    await this.customerRepository.save(connectAccount);
+
+    return connectAccount;
+  }
```
Comment (MatthewPattell) [inline microservices/payment/src/services/payment-gateway/abstract.ts]: move to separated service
microservices/payment/src/services/stripe/connect-account.ts

File: microservices/payment/src/services/payment-gateway/stripe.ts [hunk 7]
```diff
@@ -70,6 +71,19 @@ class Stripe extends Abstract {
...
+   */
+  public async createConnectAccount(userId: string): Promise<ConnectAccount> {
+    const stripeConnectAccount: StripeSdk.Account = await this.paymentEntity.accounts.create({
+      type: 'standard',
+      country: 'US',
+      email: '',
+    });
+
+    return super.createConnectAccount(userId, stripeConnectAccount.id);
+  }
```
Comment (MatthewPattell) [inline microservices/payment/src/services/payment-gateway/stripe.ts]: move to service

File: microservices/payment/src/entities/connect-account.ts [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+  @PrimaryColumn({ type: 'varchar', length: 18 })
+  @Allow()
+  connectAccountId: string;
```
Comment (MatthewPattell) [inline microservices/payment/src/entities/connect-account.ts]: microservices/microservices/payment/src/entities/customer.ts

```
interface ICustomerParams {
  accountId?: string;
}

@IsObject()
params: ICustomerParams; // json
```

**Review by MatthewPattell — CHANGES_REQUESTED** [review]

File: microservices/payment/src/methods/index.ts (hunk 2 above)
Comment (Andrey-Lavnik) [reply microservices/payment/src/methods/index.ts]: done

File: microservices/payment/src/entities/connect-account.ts (hunk 1 above)
Comment (Andrey-Lavnik) [reply microservices/payment/src/entities/connect-account.ts]: done

File: microservices/payment/src/entities/connect-account.ts (hunk 8 above)
Comment (Andrey-Lavnik) [reply microservices/payment/src/entities/connect-account.ts]: done

File: microservices/payment/src/methods/stripe/connect-account/create.ts
Comment (Andrey-Lavnik) [reply microservices/payment/src/methods/stripe/connect-account/create.ts:1]: done

File: microservices/payment/src/methods/stripe/connect-account/create.ts (hunk 3 above)
Comment (Andrey-Lavnik) [reply microservices/payment/src/methods/stripe/connect-account/create.ts]: done

File: microservices/payment/src/methods/stripe/connect-account.ts [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+class ConnectAccountInput {
+  @IsString()
+  userId: string;
+  @IsString()
+  email: string;
+  @IsString()
+  refreshUrl: string;
+  @IsString()
+  returnUrl: string;
```
Comment (MatthewPattell) [inline microservices/payment/src/methods/stripe/connect-account.ts]:
```suggestion
  @IsString()
  userId: string;
  
  @IsString()
  email: string;
  
  @IsString()
  refreshUrl: string;
  
  @IsString()
  returnUrl: string;
```

**Review by MatthewPattell — CHANGES_REQUESTED** [review]
a lot of stule guide issues

**Review by MatthewPattell — APPROVED** [review]

**Pull request merged**